source/common/router/router.h

```diff
@@ -370,19 +368,22 @@ class Filter : Logger::Loggable<Logger::Id::router>,
                                          Upstream::ResourcePriority priority) PURE;
   Http::ConnectionPool::Instance* getConnPool();
   void maybeDoShadowing();
+  bool maybeRetryReset(const absl::optional<Http::StreamResetReason>& reset_reason);
+  void onPerTryTimeout();
   void onRequestComplete();
   void onResponseTimeout();
   void onUpstream100ContinueHeaders(Http::HeaderMapPtr&& headers);
+  void onUpstreamAbort(Http::Code code, StreamInfo::ResponseFlag response_flag, std::string body, bool dropped);
```

Comment:
Its unclear from looking at the interface what this function is supposed to do, mind adding a comment?